src/flag_gems/ops/select_scatter.py

```diff
@@ -56,32 +11,20 @@ def select_scatter(inp, src, dim, index):
     assert index >= -inp.size(dim) and index < inp.size(dim), "Invalid index"
     dim = dim % inp.ndim
     index = index % inp.size(dim)
-    out = inp.clone().contiguous()
-    src = src.contiguous()
 
     valid_shape = list(inp.shape)
     del valid_shape[dim]
     assert (
         list(src.shape) == valid_shape
     ), "Expected src to have a size equal to the slice of self"
 
-    src_expanded_shape = list(inp.shape)
-    src_expanded_shape[dim] = 1
-    out_strided = restride_dim(out, dim, src_expanded_shape)
-    idx = torch.arange(0, src.numel(), device=inp.device).reshape(src_expanded_shape)
-    indices = offsetCalculator(
-        out_strided, idx, out.stride(), dim, isInp=False
-    ).squeeze(dim=dim)
-    src_offsets = offsetCalculator(src, idx, src.stride(), dim, isInp=False).squeeze(
-        dim=dim
+    out = torch.empty_strided(
+        inp.size(), inp.stride(), dtype=inp.dtype, device=inp.device
     )

```

Comment:
If the output has the same shape&stride&dtype$device as the input, it is better to use empty_like.

Reply:
`empty_like` does not keep the same stride as the input, it keeps the stride-order and allocate a dense and non-overlapping output.

Actually we always want a dense and non-overlapping output.

Reply:
I found something tricky in the output's layout.

If the input has internal overlapping, the output would not follow the input's stride, just follow the input's stride order and allocate a dense-and-non-overlapping ouput.

```python
x = torch.randn((1, 40), device="cuda").broadcast_to((3, 40))
src = torch.randn((3, 20), device="cuda") + 1000.0
out = torch.slice_scatter(x, src, dim=1, start=0, end=None, step=2)

x.shape, x.stride()
# (torch.Size([3, 40]), (0, 1))
out.shape, out.stride()
# (torch.Size([3, 40]), (40, 1))
```

```python
x = torch.randn((1, 40), device="cuda").broadcast_to((3, 40)).T
x.shape, x.stride()
# (torch.Size([40, 3]), (1, 0))

src = torch.randn((20, 3), device="cuda") + 1000.0
out = torch.slice_scatter(x, src, dim=0, start=0, end=None, step=2)
out.shape, out.stride()
# (torch.Size([40, 3]), (3, 1))
```

Reply:
So, I think it necessary to branch on whether the input has internal overlapping.

Reply:
Select_scatter have similar behavior.